wrc/codegen/merger.py

```diff
@@ -0,0 +1,177 @@
+from wrc.sema.ast import Article, Section, TableOfContent, Guideline, split_rule_number
+from copy import deepcopy
+
+NOTES_INDEX = 0
+TOC_INDEX = 1
+FIRST_ARTICLE_INDEX = 2
+
+
+class BadFormatError(Exception):
+    def __init__(self, message: str):
+        self.message = message + '. Please report this issue to the developer.'
+        super().__init__(message)
+
+
+def compare_article_numbers(num1: str, num2: str) -> bool:
+    """
+    Returns True if num1 < num2, otherwise returns False.
+    This is necessary because:
+    * We need to compare article numbers like '1' and 'A'.
+    * '2' > '10' == True, which is a problem.
+    """
+    is_digit_1 = num1.isdigit()
+    is_digit_2 = num2.isdigit()
+    if is_digit_1 and is_digit_2:
+        res = int(num1) < int(num2)
+    elif is_digit_1 and not is_digit_2:
+        # Only num2 is a letter.
+        res = True
+    elif not is_digit_1 and not is_digit_2:
+        res = num1 < num2
+    else:
+        res = False
+
+    return res
+
+
+def compare_reg_numbers(num1: list, num2: list) -> bool:
```

Comment:
From a high-level perspective it just looks like you're comparing a list of similar-ish elements (i.e. all similar types, although the exact content will of course be different in every use-case). If that is true, then I feel like there should already be a Python stdlib feature out there capable of comparing arrays.

Reply:
I couldn't find anything that also handles different list lengths this way. Something like `all(x < y for x, y in zip(list1, list2))` would be similar, but it always returns false if the lengths are different.

Reply:
["A", "12"] < ["A", "12", "c"] should be True, but the `all()` alternative returns False.